src/components/Rating.js

```diff
@@ -544,6 +502,49 @@ export default class Rating extends Component {
   }
 
   /**
+   * The function will be passed to requestAnimationFrame for touchMove
+   */
+  _triggerComponentUpdateOnTouchMove(touches) {
+    const touchedElement = document.elementFromPoint(touches.clientX, touches.clientY);
+    const value = Number(touchedElement.getAttribute('data-belle-value'));
+    if (value && this.state.focusedValue !== value) {
+      this.setState({
+        focusedValue: value
+      });
+    }
+  }
+
+
+  /**
+   * update component when component is clicked, touch ends, enter or space key are hit
+   * different update logic will apply depending on whether component has property defaultValue, value or valueLink specified
+   */
+  _triggerComponentUpdate(value) {
+    if (has(this.props, 'valueLink')) {
+      this.props.valueLink.requestChange(value);
+      this.setState({
+        focusedValue: undefined,
+        isActive: false
+      });
+    } else if (has(this.props, 'value')) {
+      this.setState({
+        focusedValue: undefined,
+        isActive: false
+      });
+    } else {
+      this.setState({
+        focusedValue: undefined,
+        isActive: false,
+        value: value
+      });
+    }
+
+    if (this.props.onUpdate) {
+      this.props.onUpdate({ value: value });
+    }
+  }
+
+  /**
```

Comment:
Would have been happy to see some strong opinion. But it was linting, good anyways :smile: